base/linalg/dense.jl

```diff
@@ -693,6 +698,198 @@ function inv(A::StridedMatrix{T}) where T
 end
 
 """
+    cos(A::Matrix)
+
+Compute the matrix cosine of a square matrix `A`.
+
+If `A` is symmetric or Hermitian, its eigendecomposition ([`eigfact`](@ref)) is
+used to compute the cosine. Otherwise, the cosine is determined by calling
+`exp`.
+
+# Examples
+```jldoctest
+julia> cos(ones(2, 2))
+2×2 Array{Float64,2}:
+  0.291927  -0.708073
+ -0.708073   0.291927
+```
+"""
+function cos(A::StridedMatrix{<:Real})
+    if issymmetric(A)
+        return full(cos(Symmetric(A)))
+    end
+    return real(exp(im*A))
+end
+function cos(A::StridedMatrix{<:Complex})
+    if ishermitian(A)
+        return full(cos(Hermitian(A)))
+    end
+    return 0.5 * (exp(im*A) + exp(-im*A))
+end
+
+"""
+    sin(A::Matrix)
+
+Compute the matrix sine of a square matrix `A`.
+
+If `A` is symmetric or Hermitian, its eigendecomposition ([`eigfact`](@ref)) is
+used to compute the sine. Otherwise, the sine is determined by calling `exp`.
+
+# Examples
+```jldoctest
+julia> sin(ones(2, 2))
+2×2 Array{Float64,2}:
+ 0.454649  0.454649
+ 0.454649  0.454649
+```
+"""
+function sin(A::StridedMatrix{<:Real})
+    if issymmetric(A)
+        return full(sin(Symmetric(A)))
+    end
+    return imag(exp(im*A))
+end
+function sin(A::StridedMatrix{<:Complex})
+    if ishermitian(A)
+        return full(sin(Hermitian(A)))
+    end
+    return -0.5im * (exp(im*A) - exp(-im*A))
+end
+
+"""
+    sincos(A::Matrix)
+
+Compute the matrix sine and cosine of a square matrix `A`.
+
+# Examples
+```jldoctest
+julia> sincos(ones(2, 2))
+([0.454649 0.454649; 0.454649 0.454649], [0.291927 -0.708073; -0.708073 0.291927])
+```
+"""
+function sincos(A::StridedMatrix{<:Real})
+    if issymmetric(A)
+        return full.(sincos(Symmetric(A)))
+    end
+    c, s = reim(exp(im*A))
+    return s, c
+end
+function sincos(A::StridedMatrix{<:Complex})
+    if ishermitian(A)
+        return full.(sincos(Hermitian(A)))
+    end
+    X, Y = exp(im*A), exp(-im*A)
+    return -0.5im * (X - Y), 0.5 * (X + Y)
```

Comment:
Same here; instead of multiplying with `-0.5im` / `0.5` here, dived by  `(-2im)` / `2` instead.